packages/node/src/breadcrumbs/FileBreadcrumbsStorage.ts

```diff
@@ -88,8 +114,15 @@ export class FileBreadcrumbsStorage implements BreadcrumbsStorage {
         };
 
         const breadcrumbJson = JSON.stringify(breadcrumb, jsonEscaper());
-        this._writer.writeLine(breadcrumbJson);
+        const jsonLength = breadcrumbJson.length + 1; // newline
+        const sizeLimit = this._limits.maximumTotalBreadcrumbsSize;
+        if (sizeLimit !== undefined) {
+            if (jsonLength > sizeLimit) {
+                return id;
+            }
```

Comment:
In the past id meant the id of the added breadcrumb. The breadcrumb id is no in the breadcrumb file. Should we return undefined instead? Or something that indicates that the breadcrumb was not added?